api/opencl_runtime_layer.asciidoc

```diff
@@ -5619,6 +5619,14 @@ buffer or an image memory object from an external handle:
     represented by its memory object, and will therefore become invalid when
     all memory objects associated with it are destroyed.
 endif::cl_khr_external_memory_win32[]
+  * {CL_EXTERNAL_MEMORY_HANDLE_OPAQUE_WIN32_NAME_KHR_anchor} specifies an NT handle name that
+    has only limited valid usage outside of OpenCL and other compatible
+    APIs.
+    It must be compatible with the functions `DuplicateHandle`,
+    `CloseHandle`, `CompareObjectHandles`, `GetHandleInformation`, and
+    `SetHandleInformation`.
+    It owns a reference to the underlying memory resource represented by its
+    memory object.
```

Comment:
I think these additions should be within the `ifdef::cl_khr_external_memory_win32` block, above.

Do we need to say anything about how the NT handle name is specified?  For reference, the Vulkan spec says for its "name" that "name is NULL or a null-terminated UTF-16 string naming the payload to import."

Presumably the property value for `CL_EXTERNAL_MEMORY_HANDLE_OPAQUE_WIN32_NAME_KHR` is a pointer to the null-terminated string.  If an application queries `CL_MEM_PROPERTIES` does it get the exact same pointer that was passed when the memory object was created, which may no longer be valid?  Or, is there an expectation that the implementation makes a copy of the name and returns it?

Same questions apply to semaphores, also.

Reply:
Regarding the query, see also: https://github.com/KhronosGroup/OpenCL-Docs/issues/396

(Discussed in the June 4th teleconference.)

Reply:
Thanks, I think this is partially fixed now.

Vulkan allows a NULL handle name, but I think this is only because the same structure is used to pass either the handle itself or the handle name.  What would it mean to pass NULL for `CL_EXTERNAL_MEMORY_HANDLE_OPAQUE_WIN32_NAME_KHR`?

Note, we do not currently allow a NULL handle for `CL_EXTERNAL_MEMORY_HANDLE_OPAQUE_WIN32_KHR` - should we?

We also still need to decide what should happen for the properties query.